packages/inertia-vue/src/form.js

```diff
@@ -0,0 +1,114 @@
+import Vue from 'vue'
+import { Inertia } from '@inertiajs/inertia'
+
+export default function(data) {
+  const defaults = JSON.parse(JSON.stringify(data))
+
+  return {
+    ...defaults,
+    errors: {},
+    hasErrors: false,
+    processing: false,
+    progress: null,
+    data() {
+      return Object
+        .keys(data)
+        .reduce((carry, key) => {
+          carry[key] = this[key]
+          return carry
+        }, {})
+    },
+    reset(fields) {
+      const args = Array.isArray(fields)
+        ? fields
+        : Object.values(arguments)
```

Comment:
Instead of using the yucky `args` variable here, couldn't we just reassign `fields`?

```suggestion
      fields = Array.isArray(fields) ? fields : Object.values(arguments)
```